packages/starknet_provider/lib/src/model/trace_transaction_trace.dart

```diff
@@ -0,0 +1,230 @@
+// ignore_for_file: invalid_annotation_target, non_constant_identifier_names
+
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:starknet/starknet.dart';
+import 'package:starknet_provider/src/model/components/event.dart';
+import 'package:starknet_provider/src/model/components/msg_to_l1.dart';
+
+import 'json_rpc_api_error.dart';
+
+part 'trace_transaction_trace.freezed.dart';
+part 'trace_transaction_trace.g.dart';
+
+class ExecuteInvocationConverter
+    implements JsonConverter<ExecuteInvocation, Map<String, dynamic>> {
+  const ExecuteInvocationConverter();
+
+  @override
+  ExecuteInvocation fromJson(Map<String, dynamic> json) {
+    if (json.containsKey('revert_reason')) {
+      return ExecuteInvocation.reverted(
+          revertReason: json['revert_reason'] as String);
+    }
+    return ExecuteInvocation.success(FunctionInvocation.fromJson(json));
+  }
+
+  @override
+  Map<String, dynamic> toJson(ExecuteInvocation object) {
+    return object.when(
+      success: (invocation) => invocation.toJson(),
+      reverted: (revertReason) => {'revert_reason': revertReason},
+    );
+  }
+}
+
+@Freezed(unionKey: 'type')
+sealed class TransactionTrace with _$TransactionTrace {
+  const factory TransactionTrace.INVOKE({
+    @JsonKey(name: 'validate_invocation')
+    FunctionInvocation? validateInvocation,
+    @JsonKey(name: 'execute_invocation')
+    @ExecuteInvocationConverter()
+    required ExecuteInvocation executeInvocation,
+    @JsonKey(name: 'fee_transfer_invocation')
+    FunctionInvocation? feeTransferInvocation,
+    @JsonKey(name: 'state_diff') TraceStateDiff? stateDiff,
+    @JsonKey(name: 'execution_resources')
+    required InnerCallExecutionResources executionResources,
```

Comment:
Hum here `execution_resources` is an `EXECUTION_RESOURCES` not an `INNER_CALL_EXECUTION_RESOURCES` (which doesn't contains `l1_data_gas` filed in 0.8.1 specification)